### ISSUE TITLE
Preprocessing data for OpenNMT baselines

### ISSUE THREAD

**Issue by Martinski95** [Alex-Fabbri/Multi-News#21]
Hi Alex, first of all, thanks for the amazing paper and dataset!

I have a question regarding preparing the data for training an OpenNMT Transformer model. I am trying to replicate the steps of first running preprocess.py, and then train.py. However, when running preprocess.py I end up with 1 train.pt, 1 valid.pt, and 1 vocab.pt files. The vocab is of size 50004, which seems odd. Afterwards, when I run the train.py, it stops at around 1300 step. I suppose this is because of the insufficient vocab size?
I tried running the preprocessing step with different versions of the data, because it was unclear to me what exactly I should use, in order to train a Transformer on the Multi-News dataset. 

The command I use for preprocessing is:
`python preprocess.py -train_src "...\train.src.cleaned" -train_tgt "...\train.tgt" -valid_src "...\val.src.cleaned" -valid_tgt "...\val.tgt" -save_data "...\newser" -src_seq_length 1000 -tgt_seq_length 1000 -src_seq_length_trunc 500 -tgt_seq_length_trunc 300 -dynamic_dict -share_vocab -shard_size 10000000`

And for training: 
`python train.py -save_model "...\transformer_1 -data ".../newser" -copy_attn -word_vec_size 512 -rnn_size 512 -layers 4 -encoder_type transformer -decoder_type transformer -position_encoding -train_steps 50000 -warmup_steps 8000 -learning_rate 2 -decay_method noam -label_smoothing 0.1 -max_grad_norm 0 -dropout 0.2 -batch_size 4096 -optim adam -adam_beta2 0.998 -param_init 0 -batch_type tokens -normalization tokens -max_generator_batches 2 -accum_count 4 -share_embeddings -param_init_glorot -seed 777 -world_size 1 -gpu_ranks 0 -save_checkpoint_steps 500
`

Would you please clarify a little bit the process of training an OpenNMT model on your data, including more information about which formats of the data are you using?

**Comment by Alex-Fabbri** [comment]
The vocabulary size there is the default for OpenNMT's preprocessing and shouldn't be the source of the problem. For the paper we used the preprocessed, truncated version which I've put online. I'm not sure why training is stopping then. Can you provide some more information about that? Is it throwing an error at that timestep?

**Comment by Martinski95** [comment]
After step 1300 or 1350 it tries to load newser.train.0.pt again, and it just hangs. I tried to save a checkpoint at step 1000 and then continue training from that checkpoint, and then again, after step 2350 it outputs `Loading dataset from ...\newser.train.0.pt` and hangs. Does it expect more than one train.pt file?

And also, I didn't quite understand if you use the preprocessed and truncated version as an input for preprocess.py? Because if I directly want to train on your preprocessed and truncated data, do I just use preprocess.py to generate the .pt files used for training?

**Comment by Alex-Fabbri** [comment]
Perhaps there is a memory problem when loading the *pt files. Maybe try playing around with max_shard_size in the preprocessing script. 
Yes, we used our preprocessed and truncated version as input to preprocess.py, so just use that to generate the .pt files.

**Comment by Martinski95** [comment]
Thanks, I reduced the max_shard_size to 10000 and now it doesn't freeze. I guess it ran out of RAM.
